plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/delete/DeleteProjectAction.java

```diff
@@ -0,0 +1,215 @@
+    private final DtoFactory                    dtoFactory;
+    private final DtoUnmarshallerFactory       dtoUnmarshaller;
+
+    @Inject
+    public DeleteProjectAction(OpenshiftLocalizationConstant locale,
+                               AppContext appContext,
+                               OpenshiftAuthorizationHandler openshiftAuthorizationHandler,
+                               DialogFactory dialogFactory,
+                               OpenshiftServiceClient service,
+                               ProjectServiceClient projectService,
+                               NotificationManager notificationManager,
+                               DtoFactory dtoFactory,
+                               DtoUnmarshallerFactory dtoUnmarshaller) {
+        super(Collections.singletonList(PROJECT_PERSPECTIVE_ID),
+              locale.deleteProjectTitle(),
+              locale.deleteProjectActionDescription(), null, null);
+
+        this.appContext = appContext;
+        this.openshiftAuthorizationHandler = openshiftAuthorizationHandler;
+        this.dialogFactory = dialogFactory;
+        this.locale = locale;
+        this.service = service;
+        this.notificationManager = notificationManager;
+        this.projectService = projectService;
+        this.dtoFactory = dtoFactory;
+        this.dtoUnmarshaller = dtoUnmarshaller;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void updateInPerspective(@NotNull ActionEvent event) {
+        CurrentProject currentProject = appContext.getCurrentProject();
+        event.getPresentation().setVisible(currentProject != null);
+        event.getPresentation().setEnabled(openshiftAuthorizationHandler.isLoggedIn()
+                                           && currentProject != null
+                                           && currentProject.getProjectDescription().getMixins().contains(OPENSHIFT_PROJECT_TYPE_ID));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        ProjectDescriptor descriptor = appContext.getCurrentProject().getProjectDescription();
+        final String nameSpace = getAttributeValue(descriptor, OPENSHIFT_NAMESPACE_VARIABLE_NAME);
+
+        Promise<List<BuildConfig>> buildConfigs = service.getBuildConfigs(nameSpace);
+        buildConfigs.then(showConfirmDialog(descriptor, nameSpace))
+                    .catchError(handleError(nameSpace));
+    }
+
+    private String getAttributeValue(ProjectDescriptor projectDescriptor, String value) {
+        return projectDescriptor.getAttributes().get(value).get(0);
+    }
+
+    private Operation<List<BuildConfig>> showConfirmDialog(final ProjectDescriptor projectDescriptor, final String nameSpace) {
+        return new Operation<List<BuildConfig>>() {
+            @Override
+            public void apply(List<BuildConfig> configs) throws OperationException {
+                if (configs.isEmpty()) {
```

Comment:
I think you should call createConfirmDialog only one time and use If structures only for defining content of label. Is it clear?